main/src/modules/contextualLayer/goalManager/src/goalManager.cpp

```diff
@@ -0,0 +1,216 @@
+#include "goalManager.h"
+#include "wrdac/subsystems/subSystem_recog.h"
+
+bool GoalManager::configure(yarp::os::ResourceFinder &rf)
+{
+    string moduleName = rf.check("name", Value("goalManager")).asString().c_str();
+    setName(moduleName.c_str());
+
+    yInfo() << moduleName << " : finding configuration files...";
+    period = rf.check("period", Value(0.1)).asDouble();
+
+    //Create an iCub Client and check that all dependencies are here before starting
+    bool isRFVerbose = false;
+    //iCub = new ICubClient(moduleName, "goalManager", "client.ini", isRFVerbose);
+    //iCub->opc->isVerbose = false;
+    //if (!iCub->connect())
+    //{
+    //    yInfo() << " iCubClient : Some dependencies are not running...";
+    //    Time::delay(1.0);
+    //}
+    rpc.open(("/" + moduleName + "/rpc").c_str());
+    attach(rpc);
+    while (!Network::connect("/ears/target:o",rpc.getName())) {
+        yWarning() << "Behavior is not reachable";
+        yarp::os::Time::delay(0.5);
+    }
+
+    portToBehavior.open("/" + moduleName + "/behavior/cmd:o");
+    while (!Network::connect(portToBehavior.getName(),"/BehaviorManager/trigger:i ")) {
+        yWarning() << "Behavior is not reachable";
+        yarp::os::Time::delay(0.5);
+    }
+    yDebug() << "Connected!";
+
+    port_behavior_context.open("/" + moduleName + "/target:o");
+    yInfo()<<"created port to behaviors";
+    
+    goal_need_port.open("/" + moduleName + "/goal:o");
+    yInfo()<<"created port to planner";
+    if (Network::connect(goal_need_port.GetName(),"/planner/rpc")) {
+        yInfo() << "planner and goalManager are connected!";
+    }
+    else { yInfo() << "planner and goalManager are NOT connected."; }
+
+    yDebug()<<"clearing bottles";
+    goal_list.clear();
+    yDebug()<<"clearing bottles";
+    current_goal = new Bottle();
```

Comment:
`current_goal` is overridden later. No need to create a new `Bottle` (and never freeing it, thus wasting memory)